lib/_http_client.js

```diff
@@ -87,6 +87,14 @@ function ClientRequest(options, cb) {
     }
     if (host && !this.getHeader('host') && setHost) {
       var hostHeader = host;
+
+      // For the Host header, ensure that IPv6 addresses are enclosed
+      // in square brackets, as defined by URI formatting
+      // https://tools.ietf.org/html/rfc3986#section-3.2.2
+      if (net.isIPv6(hostHeader)) {
```

Comment:
In my initial PR, IPv6 checking was done by `hostHeader.indexOf(':')`, just as you also suggested in #5308
However, now I agree that `net.isIPv6()` is a much more reliable way of testing for IPv6.

I don't necessarily think it's too heavyweight, since it relies on `uv_inet_pton`. I guess it's_slightly_ slower than `indexOf`, but since it also checks for valid characters inside the string, I believe it's worth the extra cycles.